statsmodels/multivariate/multivariate_ols.py

```diff
@@ -0,0 +1,480 @@
+# -*- coding: utf-8 -*-
+
+"""General linear model
+
+author: Yichuan Liu
+"""
+from __future__ import print_function, division
+
+import numpy as np
+from numpy.linalg import eigvals, inv, solve, matrix_rank, pinv, svd
+from scipy import stats
+import pandas as pd
+from patsy import DesignInfo
+from statsmodels.compat import string_types
+
+from statsmodels.base.model import Model
+from statsmodels.iolib import summary2
+__docformat__ = 'restructuredtext en'
+
+_hypotheses_doc = """
+    hypotheses: A list of tuples
+        Hypothesis to be tested. Each element is an array-like
+                            (name, contrast_L, transform_M)
+        containing a string `name`, the contrast matrix L and the transform
+        matrix M (for transforming dependent variables), respectively.
+        contrast_L : array-like or an array of strings
+           Contrast matrix for hypotheses testing.
+           If array-like, each row is an hypotheses and each column is an
+           independent variable. At least 1 row
+           (1 by k_exog, the number of independent variables) is required.
+           If an array of strings, it will be passed to
+           patsy.DesignInfo().linear_constraint.
+        transform_M : array-like or an array of strings
+            Transform matrix.
+            Default to be k_endog by k_endog identity matrix
+            (i.e. do not transform y matrix).
+           If an array of strings, it will be passed to
+           patsy.DesignInfo().linear_constraint.
+        If `hypotheses` is None: 1) the effect of each independent variable
+        on the dependent variables will be tested. Or 2) if model is created
+        using a formula,  `hypotheses` will be created according to
+        `design_info`. 1) and 2) is equivalent if no additional variables
+        are created by the formula (e.g. dummy variables for categorical
+        variables and interaction terms)
+"""
+
+
+def _multivariate_ols_fit(x, y, method='svd', tolerance=1e-8):
```

Comment:
_multivariate_ols_fit(endog, exog)
you can use y, x internally as shortcut alias if needed